src/model_warehouse/resources.py

```diff
@@ -82,14 +83,20 @@ class IndvModel(Resource):
     def get(self, id):
         """Return a model by ID."""
         app.logger.debug("Fetching model by ID {}".format(id))
-        return Model.query.filter(Model.id == id).one()
+        try:
```

Comment:
It very much looks like the block

```python
        try:
            return Model.query.filter(Model.id == id).one()
        except NoResultFound:
            abort(404, f"Cannot find any model with id {id}")
```

should be abstracted into a utility method to be used by all the RUD methods.

Reply:
Seems a bit overkill to me. There's nothing unreadable about this simple logic, and abstracting it away would only decrease readability in my opinion.

Reply:
```python
    @staticmethod
    def fetch_model(id: int) -> Model:
        app.logger.debug("Fetching model by ID %d.", id)
        try:
            return Model.query.filter(Model.id == id).one()
        except NoResultFound:
            abort(404, f"Cannot find any model with id {id}.")
```
and then, for example, `delete` becomes simply:

```python
    def delete(self, id):
        """Delete a model by ID."""
        model = self.fetch_model(id)
        db.session.delete(model)
        db.session.commit()
        return model
```
I find that very easy to grok and then I know it's the same piece of code in the other methods immediately.

I also realized that it could be nice to respond with more specific return codes, [this API tutorial](https://www.restapitutorial.com/lessons/httpmethods.html) seems quite reasonable to me.